ansible/roles/controller/tasks/deploy.yml

```diff
@@ -292,6 +292,19 @@
     env: "{{ env | combine(elastic_env) }}"
   when: db.activation_store.backend == "ElasticSearch"
 
+- name: setup mongodb artifact store env
+  set_fact:
+    mongodb_env:
+      "CONFIG_whisk_mongodb_uri": "{{ db.mongodb.connect_string }}"
+      "CONFIG_whisk_mongodb_database": "{{ db.mongodb.database }}"
+      "CONFIG_whisk_spi_ArtifactStoreProvider": "org.apache.openwhisk.core.database.mongodb.MongoDBArtifactStoreProvider"
+  when: db.backend == "MongoDB"
```

Comment:
What happens if this is defined along with `db.activation_store.backend == "ElasticSearch"`?

I feel we need to consolidate DB related configuration.

Reply:
it should work well with ES as activation backend while mongodb store other entities

Reply:
I meant we need a consolidated way to configure multiple DBs for each entity store as we have more options for them.

For example, we might consider the following option:

1. Use the database for all entities if only `db.backend` is defined.
2. Use a different database for activation if `db.activation_store.backend` is defined along with `db.backend`.
3. Use a different database for whisk entities if `db.artifact_store.backend` is defined along with `db.backend`.
.
.
.

if no specific backend store is defined, we can fall back to `db.backend` by default.
We can have fine granularity but I feel like having two configurations for activations and the others would be enough for now?

Reply:
I think `activation` is a kind of `artifact`? so `db.artifact_store.backend` should refer to all kind of entities including `activation`, which is just like what `db.backend` do here

Reply:
I think we can name them accordingly as long as we can selectively configure activation stores and the other stores.

Reply:
so use `db.artifact_store.backend` for all entities, and `db.activation_store.backend` for activations?

Reply:
That would be a good starting point.
I believe we can add more options for subjects, whisk entities in the future if required.